plugins/connection/httpapi.py

```diff
@@ -142,6 +142,16 @@
     - name: ANSIBLE_BECOME_METHOD
     vars:
     - name: ansible_become_method
+  platform_type:
+    description:
+    - Set type of platform.
+    ini:
```

Comment:
The ini option doesn't work when setting option from action plugin, so IMO it can be removed for now till a workaround is in place. The rest of the PR looks good to me.

Reply:
+1